File: src/main/java/io/github/mschout/gitlab/toggltimer/configuration/JacksonConfiguration.java
package io.github.mschout.gitlab.toggltimer.configuration;

import org.springframework.boot.jackson.autoconfigure.JsonMapperBuilderCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import tools.jackson.databind.cfg.DateTimeFeature;

@Configuration
public class JacksonConfiguration {

  // serialize Instant as ISO-8601 string
  @Bean
  JsonMapperBuilderCustomizer jacksonCustomizer() {
    return builder -> builder.disable(DateTimeFeature.WRITE_DATES_AS_TIMESTAMPS);
  }
}

File: src/main/kotlin/io/github/mschout/gitlab/toggltimer/configuration/JacksonConfiguration.kt
package io.github.mschout.gitlab.toggltimer.configuration

import org.springframework.boot.jackson.autoconfigure.JsonMapperBuilderCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import tools.jackson.databind.cfg.DateTimeFeature

@Configuration
class JacksonConfiguration {
  @Bean
  fun jacksonCustomizer(): JsonMapperBuilderCustomizer = JsonMapperBuilderCustomizer { builder ->
    builder.disable(DateTimeFeature.WRITE_DATES_AS_TIMESTAMPS)
  }
}
